### DISCUSSION TITLE
Do we know why I am seeing 200 on OPTION request but 500 on POST while querying edge functions?

### DISCUSSION THREAD

**Discussion by anup-khanal** [supabase/supabase#21458] [Questions]
I am not able to call supabase functions. OPTION works just fine but POST doesn't.

```
Log ID
67106cb7-07e6-46c8-835a-4f802b3c1da0

Log Timestamp (UTC)
2024-02-22T17:42:13.041Z

Log Event Message
POST | 500 | https://XXXXXXXXXXXXXX.supabase.co/functions/v1/ask

Log Metadata
[
  {
    "deployment_id": "XXXXXXXXXXXXXX_229a3dba-87ca-4cb5-8f3b-7a4fb76ef9e4_1",
    "execution_time_ms": 474,
    "function_id": "229a3dba-87ca-4cb5-8f3b-7a4fb76ef9e4",
    "project_ref": "XXXXXXXXXXXXXX",
    "request": [
      {
        "headers": [
          {
            "accept": "*/*",
            "accept_encoding": "gzip",
            "connection": "Keep-Alive",
            "content_length": "47",
            "cookie": null,
            "host": "XXXXXXXXXXXXXX.supabase.co",
            "user_agent": "PostmanRuntime/7.36.1"
          }
        ],
        "host": "XXXXXXXXXXXXXX.supabase.co",
        "method": "POST",
        "pathname": "/functions/v1/ask",
        "port": null,
        "protocol": "https:",
        "search": null,
        "url": "https://XXXXXXXXXXXXXX.supabase.co/functions/v1/ask"
      }
    ],
    "response": [
      {
        "headers": [
          {
            "content_length": "41",
            "content_type": "text/plain;charset=UTF-8",
            "date": "Thu, 22 Feb 2024 17:42:13 GMT",
            "server": "cloudflare",
            "vary": "Accept-Encoding",
            "x_sb_edge_region": "us-east-1",
            "x_served_by": "supabase-edge-runtime"
          }
        ],
        "status_code": 500
      }
    ],
    "version": "1"
  }
]

```

**Comment by GaryAustin1** [comment]
Does not seem like enough info for someone to help you.  What debug have you done on your function?  Any console logs with info?  What is the function doing and would it pass back errors from that?

Could be lots of things.
One thread:  https://github.com/supabase/supabase/issues/18043

**Reply by anup-khanal** [reply]
this is how I am calling the function and I am passing the question param as a body. 
```
await supabaseClient.functions.invoke(
    "ask",
    {
      body: JSON.stringify({ query: question }),
    }
  );
```
This is what I see in the console. 
![Screenshot 2024-02-22 at 6 35 10 PM](https://github.com/supabase/supabase/assets/32517600/05213ed5-24bc-4872-83f1-ec24d68cd28f)

I am doing below and this is what you see in the picture above
```
console.log(data);
console.log(error);
```

**Reply by GaryAustin1** [reply]
I don't think you are supposed to stringify when using invoke. https://supabase.com/docs/reference/javascript/functions-invoke?example=error-handling

**Reply by anup-khanal** [reply]
I tried without stringify and it still does not work. If it was an issue with the request, wouldn't it say 400 Bad Request or something?

**Reply by anup-khanal** [reply]
I even tried using postman. Seeing Internal Server Error.
<img width="1345" alt="Screenshot 2024-02-22 at 8 29 24 PM" src="https://github.com/supabase/supabase/assets/32517600/9f8d086f-4424-4bdf-a5a0-48896bac7c7c">

**Reply by GaryAustin1** [reply]
Did you read thru my original link and authorization header?

Do you have console.logs in the function itself and do they run?

**Reply by anup-khanal** [reply]
I have the bearer auth token in the call if that is what you are asking.

<img width="1293" alt="Screenshot 2024-02-22 at 8 48 37 PM" src="https://github.com/supabase/supabase/assets/32517600/0da7b320-b263-456c-8d5e-99afb55063ca">

And for console log, are you referring to this?

```
export async function askQuestion(question) {
  const { data, error } = await supabaseClient.functions.invoke(
    "ask-custom-data",
    {
      body: { query: question },
    }
  );
  console.log(data);
  console.log(error);
}
```

**Reply by GaryAustin1** [reply]
No, In the edge function itself.

I mainly was getting you to post more info as your first post likely would not help anyone.  I'm at my limit on what I know about edge functions.  500 is very generic and without more info on what is causing it hard to help.  Also the link I gave showed how to get errors back from the edge function:  https://supabase.com/docs/reference/javascript/functions-invoke?example=error-handling

**Reply by anup-khanal** [reply]
I think I got it resolved. Here is few things to take note of, for anyone out there,

I had to specify these in the header for POST:

```
headers: {
      "Content-Type": "application/json",
      "Access-Control-Allow-Origin": "*",
      "Access-Control-Allow-Methods": "POST",
      "Access-Control-Expose-Headers": "Content-Length, X-JSON",
      "Access-Control-Allow-Headers": "apikey,X-Client-Info, Content-Type, Authorization, Accept, Accept-Language, X-Authorization",
    }
```

Also the open API model that I was using was old. Was using `text-embedding-ada-002` and is now replaced with `gpt-3.5-turbo-instruct`. This was causing the function to fail as well.

FYI: https://stackoverflow.com/questions/77789886/openai-api-error-the-model-text-davinci-003-has-been-deprecated

Thanks @GaryAustin1 for your help. We can close this discussion.
